batch/batch/cloud/gcp/driver/resource_manager.py

```diff
@@ -0,0 +1,133 @@
+            if e.status == 404:
+                raise VMDoesNotExist() from e
+            raise
+
+    async def get_vm_state(self, instance: Instance) -> VMState:
+        try:
+            spec = await self.compute_client.get(f'/zones/{instance.location}/instances/{instance.name}')
+            state = spec['status']  # PROVISIONING, STAGING, RUNNING, STOPPING, TERMINATED
+
+            if state in ('PROVISIONING', 'STAGING'):
+                return VMStateCreating(spec, instance.time_created)
+            if state == 'RUNNING':
+                last_start_timestamp_msecs = parse_gcp_timestamp(spec.get('lastStartTimestamp'))
+                assert last_start_timestamp_msecs is not None
+                return VMStateRunning(spec, last_start_timestamp_msecs)
+            if state in ('STOPPING', 'TERMINATED'):
+                return VMStateTerminated(spec)
+            log.exception(f'Unknown gce state {state} for {instance}')
+            return UnknownVMState(spec)
+
+        except aiohttp.ClientResponseError as e:
+            if e.status == 404:
+                raise VMDoesNotExist() from e
+            raise
+
+    def machine_type(self, cores: int, worker_type: str) -> str:
+        return family_worker_type_cores_to_gcp_machine_type(
+            GCP_MACHINE_FAMILY, worker_type, cores)
+
+    def worker_type_and_cores(self, machine_type: str) -> Tuple[str, int]:
+        return gcp_machine_type_to_worker_type_and_cores(machine_type)
+
+    def instance_config(self,
+                        machine_type: str,
+                        preemptible: bool,
+                        local_ssd_data_disk: bool,
+                        data_disk_size_gb: int,
+                        boot_disk_size_gb: int,
+                        job_private: bool,
+                        ) -> GCPSlimInstanceConfig:
+        return GCPSlimInstanceConfig(
+            machine_type,
+            preemptible,
+            local_ssd_data_disk,
+            data_disk_size_gb,
+            boot_disk_size_gb,
+            job_private,
+        )
+
+    def instance_config_from_dict(self, data: dict) -> GCPSlimInstanceConfig:
+        return GCPSlimInstanceConfig.from_dict(data)
+
+    async def create_vm(self,
+                        file_store: FileStore,
+                        resource_rates: Dict[str, float],
+                        machine_name: str,
+                        activation_token: str,
+                        max_idle_time_msecs: int,
+                        local_ssd_data_disk: bool,
+                        data_disk_size_gb: int,
+                        boot_disk_size_gb: int,
+                        preemptible: bool,
+                        job_private: bool,
+                        location: str,
+                        machine_type: str,
+                        instance_config: InstanceConfig,
+                        ) -> List[Dict[str, Any]]:
+        if local_ssd_data_disk:
+            assert data_disk_size_gb == 375
```

Comment:
Did you change the semantics here? It should be 0.

Reply:
The term "data_disk_size_gb" now means the size of the data disk, regardless of its type. I did this so I could centralize the definition of how big a local ssd is [in one place](https://github.com/hail-is/hail/pull/10920/files#diff-b3aa93ab91e349f8d38b2c10efeddd451eef1e1f884bb8d5527ed96cb19bf083R89). I think I still need to push that through everywhere. I saw a lot of code that was like this:
```
if is_local_ssd:
    return XYZ 375 ABC
else:
    return XYZ pd_ssd_size ABC
```
which said to me that there was a concept, data disk size, which wasn't represented directly in the code.

Reply:
I'll push it through fully in a separate PR. There's enough changes in here already.